website/source/docs/providers/aws/r/elb.html.markdown

```diff
@@ -10,6 +10,12 @@ description: |-
 
 Provides an Elastic Load Balancer resource.
 
+~> **NOTE on ELB Instances and ELB Attachments:** Terraform currently
+provides both a standalone [ELB Attachment resource](elb_attachment.html)
+(describing the instances attached to an ELB), and an ELB resource with
+`instances` defined in-line. At this time you cannot use an ELB with in-line
+instaces in conjunction with an ELB Attachment resource. Doing so will cause a
+conflict of rule settings and will overwrite rules.
```

Comment:
That said user still has to define at least 1 listener inside `aws_elb` as ELBs without listeners are disallowed on the API level 😢 

I'm wondering if it's better to implement the `aws_elb_attachment` as a data source that would be somehow folded into `aws_elb.dynamic_listeners`. I haven't looked into how the data sources work under the hood as yet, but reversing the dependency (ELB / listeners) here would IMO help as all listeners would be defined in a single place and generating diff/plan would be easier I think.

Also we could then mark `aws_elb.listener` as `ConflictsWith` `dynamic_listeners` which would prevent users from doing stupid things.